common/persistence/sql/sqlplugin/sqlite/plugin.go

```diff
@@ -59,11 +63,14 @@ func (p *plugin) CreateDB(
 	cfg *config.SQL,
 	r resolver.ServiceResolver,
 ) (sqlplugin.DB, error) {
-	conn, err := p.createDBConnection(cfg, r)
+	conn, err := p.connPool.Allocate(cfg, r, p.createDBConnection)
 	if err != nil {
 		return nil, err
 	}
+
 	db := newDB(dbKind, cfg.DatabaseName, conn, nil)
+	db.OnClose(func() { p.connPool.Close(cfg) }) // remove reference
```

Comment:
does it make sense to set this hook in Allocate() method so caller does not need to know about it and just call Close on the conn.

Reply:
Possibly, I’ll check this option. The problem is that we need temporary DB overlay for migration, I hope it won’t trigger the Close too early.

Reply:
Then pool has to produce DB instances, not sdlx.DB. I've been thinking about ut, but in the current flow, it won't help anyway since refCount does not properly with in-memory storage.

Reply:
I see. I'm ok with current implementation.